### PR TITLE
[IA] Rename /docs/instrumentation to /docs/languages, and Manual subpages to Instrumentation

### PR DESCRIPTION
**Pull request by svrnm**
I was once again trying to wrap my head around #3228, and I wanted to start with a first part of what I have proposed [in this comment](https://github.com/open-telemetry/opentelemetry.io/issues/3228#issuecomment-1711388378):

* Replace top level section "Instrumentation" with "Language APIs & SDKs". This (imho) makes it also easier for people to find what they are looking for if they are not yet firm in the wording (aka they don't know what "Instrumentation" means)
* Replace "Manual" with "Instrumentation" because this guide is going in detail on how to instrument your code (here I assume that someone went through the getting started and is already aware of what "Instrumentation" is)

This PR purposefully is not yet touching the word "Automatic", this is worth a separate consideration

Final Note, I kick this off as a draft PR for discussion, because right now this PR only changes `linkTitle`s and if we want to go with this change, we need to rename files & URLs and have to manage a ton of aliases.

@open-telemetry/docs-approvers PTAL

**Preview**: https://deploy-preview-3761--opentelemetry.netlify.app/docs/languages/

**Redirect tests** (selected):

- https://deploy-preview-3761--opentelemetry.netlify.app/docs/instrumentation/
- https://deploy-preview-3761--opentelemetry.netlify.app/docs/instrumentation/go/
- https://deploy-preview-3761--opentelemetry.netlify.app/docs/instrumentation/js/getting-started/nodejs/
- https://deploy-preview-3761--opentelemetry.netlify.app/docs/instrumentation/java/manual/
- https://deploy-preview-3761--opentelemetry.netlify.app/docs/instrumentation/php/manual/

Screenshot 2024-01-12 at 16 22 57

### CODE REVIEW

**Review by theletterf — COMMENTED** [review]
Sounds reasonable, as it leaves the door open to further variants, such as "automatic instrumentation". I do wonder, though: Isn't this sort of assuming that the code based instrumentation is the default way of instrumenting? That might not be true for runtime languages like Node or Java.

I like @pellared 's structure here, which might inform a deeper restructuring of our docs: https://github.com/open-telemetry/opentelemetry.io/issues/3228#issuecomment-1708066921

**Comment by svrnm** [comment]
> Sounds reasonable, as it leaves the door open to further variants, such as "automatic instrumentation". I do wonder, though: Isn't this sort of assuming that the code based instrumentation is the default way of instrumenting? That might not be true for runtime languages like Node or Java.

I'll make a strong statement now and say, yes, code based instrumentation is the default way, because [telemetry should be built-in](https://github.com/open-telemetry/opentelemetry.io/issues/3228#issuecomment-1708066921), so agents and instrumentation libraries are only an intermediate thing until otel is native everywhere. We see this actually happening for Java with the Spring Boot Starter and for Node.Js with nextjs having otel built-in natively (and by that not needing any instrumentation library anymore)

> I like @pellared 's structure here, which might inform a deeper restructuring of our docs: [#3228 (comment)](https://github.com/open-telemetry/opentelemetry.io/issues/3228#issuecomment-1708066921)

I was thinking about this as well, because "Instrumentation" right now also includes "Bootstrap/Setup the SDK" right now, which probably should be extracted (maybe in it's own PR). We have something for "How you "instrument libraries" which is called "Libraries" in some of the languages.

File: netlify.toml [hunk 1]
```diff
@@ -15,6 +15,11 @@ from = "https://blog.opentelemetry.io/*"
 to = "https://opentelemetry.io/blog/:splat"
 force = true
 
+[[redirects]]
+from = "/docs/instrumentation/*"
+to = "/docs/languages/:splat"
+force = true
```
Comment (chalin) [inline netlify.toml:21]: As discussed in Slack, add this rule to the new languages index file instead. Follow the Registry page as an example.

**Review by chalin — COMMENTED** [review]
I like this! 👍🏻

File: netlify.toml (hunk 1 above)
Comment (chalin) [reply netlify.toml:21]: Add the following to the Languages index front matter:

```yaml
aliases: [/docs/instrumentation]
redirects: [{ from: /docs/instrumentation/*, to: ':splat' }]
```

**Comment by cartermp** [comment]
I like the concept of the change. I'm happy to proceed with this moving out of a draft for review.

File: netlify.toml (hunk 1 above)
Comment (svrnm) [reply netlify.toml]: thanks!

**Comment by pellared** [comment]
I think it is a great improvement.

**Review by Kielek — APPROVED** [review]

**Review by theletterf — APPROVED** [review]

**Review by brettmc — APPROVED** [review]

**Review by cartermp — APPROVED** [review]

**Comment by svrnm** [comment]
> The main thing in my mind here is:
> 
> > This PR purposefully is not yet touching the word "Automatic", this is worth a separate consideration
> 
> It looks a little odd without some tweaks to both the automatic sections and the (if relevant) libraries sections. That's a separate issue to tackle, but it needs tackling.

Agreed, we need to review the other section titles as well! Let's have some follow up PRs then

**Review by chalin — CHANGES_REQUESTED** [review]
Oops, the redirects for `manual` pages aren't implemented correctly. Let me see how we can fix this.

**Comment by chalin** [comment]
@svrnm - actually, if you'd like to merge this now, I can fix the redirects in a followup PR. WDYT?

**Comment by svrnm** [comment]
> @svrnm - actually, if you'd like to merge this now, I can fix the redirects in a followup PR. WDYT?

as discussed via slack I try to fix them right now because there are a few other things that need to be updated (links on the status page)

File: content/en/docs/languages/go/instrumentation.md [hunk 2]
```diff
@@ -1,15 +1,15 @@
 ---
-title: Manual Instrumentation
-linkTitle: Manual
+title: Instrumentation
+linkTitle: Instrumentation
```
Comment (chalin) [inline content/en/docs/languages/go/instrumentation.md:3]:
```suggestion
```

File: content/en/docs/languages/java/instrumentation.md [hunk 3]
```diff
@@ -1,43 +1,44 @@
 ---
-title: Manual Instrumentation
-linkTitle: Manual
+title: Instrumentation
+linkTitle: Instrumentation
```
Comment (chalin) [inline content/en/docs/languages/java/instrumentation.md:3]:
```suggestion
```

File: content/en/docs/languages/net/instrumentation.md [hunk 4]
```diff
@@ -1,11 +1,12 @@
 ---
-title: Manual Instrumentation
-linkTitle: Manual
-weight: 30
-description: Manual instrumentation for OpenTelemetry .NET
+title: Instrumentation
+linkTitle: Instrumentation
```
Comment (chalin) [inline content/en/docs/languages/net/instrumentation.md:3]:
```suggestion
```

File: content/en/docs/languages/php/instrumentation.md [hunk 5]
```diff
@@ -1,14 +1,15 @@
 ---
-title: Manual Instrumentation
-linkTitle: Manual
-weight: 30
+title: Instrumentation
+linkTitle: Instrumentation
```
Comment (chalin) [inline content/en/docs/languages/php/instrumentation.md:3]:
```suggestion
```

File: content/en/docs/languages/python/instrumentation.md [hunk 6]
```diff
@@ -1,14 +1,15 @@
 ---
-title: Manual Instrumentation
-linkTitle: Manual
+title: Instrumentation
+linkTitle: Instrumentation
```
Comment (chalin) [inline content/en/docs/languages/python/instrumentation.md:3]:
```suggestion
```

File: content/en/docs/languages/ruby/instrumentation.md [hunk 7]
```diff
@@ -1,16 +1,17 @@
 ---
-title: Manual Instrumentation
-linkTitle: Manual
+title: Instrumentation
+linkTitle: Instrumentation
```
Comment (chalin) [inline content/en/docs/languages/ruby/instrumentation.md:3]:
```suggestion
```

File: content/en/docs/languages/swift/instrumentation.md [hunk 8]
```diff
@@ -1,11 +1,12 @@
 ---
-title: Manual Instrumentation
-linkTitle: Manual
+title: Instrumentation
+linkTitle: Instrumentation
```
Comment (chalin) [inline content/en/docs/languages/swift/instrumentation.md:3]:
```suggestion
```

**Review by chalin — APPROVED** [review]
LGTM. In particular the redirects are fixed.

See inline for minor suggestions.

**Pull request merged**